src/app/core/services/rundown-timing-context-state.service.ts

```diff
@@ -0,0 +1,187 @@
+    this.stopTimeResolutionTimer()
+    this.isHighResolutionTimer = true
+    this.timeResolutionTimerId = setInterval(this.onTimeTick.bind(this), HIGH_RESOLUTION_INTERVAL_DURATION_IN_MS)
+  }
+
+  private onTimeTick(): void {
+    this.rundowns.forEach(rundown => {
+      const rundownTimingContextSubject: BehaviorSubject<RundownTimingContext> | undefined = this.getRundownTimingContextSubject(rundown.id)
+      if (!rundownTimingContextSubject) {
+        return
+      }
+      const currentEpochTime: number = Date.now()
+      const previousRundownTimingContext: RundownTimingContext = rundownTimingContextSubject.value
+      const expectedStartEpochTimeForRundown: number = this.rundownTimingService.getExpectedStartEpochTimeForRundown(
+        rundown,
+        previousRundownTimingContext.expectedDurationInMsForRundown,
+        currentEpochTime
+      )
+      const expectedEndEpochTimeForRundown: number = this.rundownTimingService.getExpectedEndEpochTimeForRundown(rundown, previousRundownTimingContext.expectedDurationInMsForRundown, currentEpochTime)
+      const playedDurationInMsForOnAirPart: number = this.rundownTimingService.getPlayedDurationInMsForOnAirPart(rundown, currentEpochTime)
+      const playedDurationInMsForOnAirSegment: number = this.rundownTimingService.getPlayedDurationInMsForOnAirSegment(rundown, currentEpochTime)
+      const durationInMsSpentInOnAirSegment: number = this.rundownTimingService.getDurationInMsSpentInOnAirSegment(rundown, currentEpochTime)
+      const remainingDurationInMs: number = this.rundownTimingService.getRemainingDurationInMsForRundown(
+        rundown,
+        previousRundownTimingContext.expectedDurationsInMsForSegments,
+        playedDurationInMsForOnAirSegment
+      )
+      const rundownTimingContext: RundownTimingContext = {
+        ...rundownTimingContextSubject.value,
+        currentEpochTime,
+        expectedStartEpochTimeForRundown,
+        expectedEndEpochTimeForRundown,
+        playedDurationInMsForOnAirPart,
+        playedDurationInMsForOnAirSegment,
+        durationInMsSpentInOnAirSegment,
+        remainingDurationInMsForRundown: remainingDurationInMs,
+      }
+      rundownTimingContextSubject.next(rundownTimingContext)
+    })
+
+    const hasActiveRundown: boolean = [...this.rundowns.values()].some(rundown => rundown.isActive)
+    if (hasActiveRundown && !this.isHighResolutionTimer) {
+      this.startHighResolutionTimer()
+    } else if (!hasActiveRundown && this.isHighResolutionTimer) {
+      this.startLowResolutionTimer()
+    }
+  }
+
+  private startLowResolutionTimer(): void {
+    this.stopTimeResolutionTimer()
+    this.isHighResolutionTimer = false
+    this.timeResolutionTimerId = setInterval(this.onTimeTick.bind(this), LOW_RESOLUTION_INTERVAL_DURATION_IN_MS)
+  }
+
+  private stopTimeResolutionTimer(): void {
+    if (this.timeResolutionTimerId === undefined) {
+      return
+    }
+    this.isHighResolutionTimer = false
```

Comment:
This seems unnecessary. Both your "startTimers" set this when you start them.
This is only necessary because of your logic in `onTimeTick()`.

Reply:
Is removed.